sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -980,6 +1069,9 @@ private Channel pollChannel() {
     private void releaseAndOfferChannel(final Channel channel, final Promise<Void> promise) {
         this.ensureInEventLoop();
         try {
+
+            // TODO: moderakh is this right?!!!!!
+            this.acquiredChannels.remove(channel);
```

Comment:
Did it help in any scenario or is it a defensive programming?

Reply:
Yes - removed the comment. There was one scenario where we ended up leaving a channel in acquiredChannels and availableChannels. This was part of the fix to make sure we don't establish more connections than MaxChannels...